### PR TITLE
Fix bug for adding multi period ws

### LINKED ISSUE #17815: Bug when adding multi-period event data in ISIS SANS
Rob wrote:

> I’ve just tried Adding runs 13065 and 13057 using the GUI but I get an error because of workspace groups.

It appears that the `getRun` method is required which GroupWorkspaces don't have. This means we need to make sure to check if we are dealing with a GroupWorkspace or a regular workspace and if we are dealing with a GroupWorkspace we check the first child.

### PR DESCRIPTION
**Pull request by AntonPiccardoSelg**
Fixes #17815

Added multi-period event workspaces were not supported up until now. Rob needs them. This PR makes them available.

Don't merge before #17910 is merged.

**To test:**

**Adding multiperiod event data as event**

The relevant files canbe found here: \\olympic\babylon5\Scratch\Anton\Testing\17890 _Fix_bug_for_adding_multi_period_ws\Adding multiperiod event data as histogram
Set the Mantid path and the save path to this directory.

1. Open the ISIS SANS GUI and set the instrument to LARMOR
1. Load the user file: USER_Larmor_163F_HePATest_r13038.txt
1. Go to the `Add Runs ` tab 
1. Add LARMOR00013065.nxs and LARMOR00013057.nxs to the list of files to add.
1. Select `Save as event data`
1. Press sum
  * Confirm that there are no error messages and that you get a pop up message confirming that the files were added and stored in a file called xxxxx-add.nxs
1. Go to the `Run Numbers` tab
1. For the scatter sample select the xxxxx-add.nxs file. 
1. Press 1DReduce.
  * Confirm that the reduction happens without any issues.

**Adding multiperiod event data as histogram**

Follow the same steps as in **Adding multiperiod event data as event**. But instead of selecting  `Save as event data` select `Use binning from monitors`.

**Adding histogram data** 

Run the steps from above but use `LARMOR00000063.nxs` twice for the addition. You won't be able to set anything for the histogram binning because we are using a histogram workspace here,

**Adding Workspace2D multiperiod data**

The relevant files canbe found here: \\olympic\babylon5\Scratch\Anton\Testing\17890 _Fix_bug_for_adding_multi_period_ws\Adding Workspace2D multiperiod data
Set the Mantid path and the save path to this directory.

1. Open the ISIS SANS GUI and set the instrument to SANS2D
1. Load the user file: MASKSANS2D.091A
1. Go to the `Add Runs ` tab 
1. Add SANS2D0005512.nxs and SANS2D0005512.nxs to the list of files ( we add it to itself)
1. Press sum
  * Confirm that there are no error messages and that you get a pop up message confirming that the files were added and stored in a file called xxxxx-add.nxs
1. Go to the `Run Numbers` tab
1. For the scatter sample select the xxxxx-add.nxs file. 
1. Press 1DReduce.
  * Confirm that the reduction happens without any issues.

**Adding event data**

They are quite heavily covered in the system tests.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by AntonPiccardoSelg** [comment]
@lottiegreenwood  I have placed a Windows installer here: \\olympic\babylon5\Scratch\Anton\Testing\17890 _Fix_bug_for_adding_multi_period_ws

**Comment by lottiegreenwood** [comment]
works as described :shipit:

File: MantidQt/CustomInterfaces/inc/MantidQtCustomInterfaces/SANSRunWindow.ui [hunk 1]
```diff
@@ -93,7 +93,7 @@
        <string notr="true"/>
       </property>
       <property name="currentIndex">
-       <number>0</number>
+       <number>5</number>
```
Comment (martyngigg) [inline MantidQt/CustomInterfaces/inc/MantidQtCustomInterfaces/SANSRunWindow.ui]: Just wanted to check this. I think this change sets the default tab index to `5`, doesn't it? Is that intended?

**Comment by AntonPiccardoSelg** [comment]
@martyngigg Thanks for spotting the current index issue. This has been reverted back.

**Pull request merged**